src/ConnectFour/Port/Adapter/Persistence/Repository/PredisGameStore.php

```diff
@@ -28,12 +35,25 @@ public function find(GameId $gameId): Game
         return $storedGame ? $this->deserializeGame($storedGame) : $this->fallbackGameFinder->find($gameId);
     }
 
-    public function save(Game $game): void
+    public function persist(Game $game): void
     {
-        $this->predis->set(
-            $this->storageKeyPrefix . $game->id(),
-            $this->serializeGame($game)
-        );
+        $this->pendingGames[$game->id()] = $game;
```

Comment:
This is already an optimization. The key should be removed during performance testing to have a fair comparison with the old implementation.